cmd/network-slirp-binding/server/server.go

```diff
@@ -58,7 +63,17 @@ type V1alpha2Server struct {
 func (s V1alpha2Server) OnDefineDomain(_ context.Context, params *hooksV1alpha2.OnDefineDomainParams) (*hooksV1alpha2.OnDefineDomainResult, error) {
 	log.Log.Info("OnDefineDomain callback method has been called")
 
-	newDomainXML, err := callback.OnDefineDomain(params.GetDomainXML(), nil)
+	vmi := &vmschema.VirtualMachineInstance{}
+	if err := json.Unmarshal(params.GetVmi(), vmi); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal VMI: %v", err)
+	}
+
+	slirpConfigurator, err := domain.NewSlirpNetworkConfigurator(vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, s.SearchDomains)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create slirp configurator: %v", err)
```

Comment:
It seems it will fail if no slirp interface is detected.
Should we really fail or just log an error on stderr?

Reply:
IMO it should fail, it means that the plugin is called with invalid VM configuration that should have been detected by the webhook.

Reply:
It also means that the sidecar runs even if slirp is not defined on the VMI spec.
Maybe someone defined in manually to run and not through the new API.

@AlonaKaplan , what do you think?

Reply:
Seems reasonable to fail. The sidecar should work only with the new API, if somebody configure it manually, it makes sense it will fail...